swarm.go

```diff
@@ -374,11 +374,24 @@ func (s *Swarm) Start(srv *p2p.Server) error {
 	log.Info("Updated bzz local addr", "oaddr", fmt.Sprintf("%x", newaddr.OAddr), "uaddr", fmt.Sprintf("%s", newaddr.UAddr))
 
 	if s.config.SwapEnabled {
-		err := s.swap.Deploy(context.Background(), s.backend, s.config.Path)
-		if err != nil {
-			return fmt.Errorf("Unable to deploy swap contract: %v", err)
+		if s.config.Contract != (common.Address{}) {
+			address := s.config.Contract
+			err := cswap.ValidateCode(context.Background(), s.backend, address)
+			if err != nil {
+				return fmt.Errorf("Provided address not a chequebook smart-contract: %v", err)
+			}
+			err = s.swap.NewInstanceAt(address, s.backend)
+			if err != nil {
+				return fmt.Errorf("Could not set the instance at provided cheqeubook: %v", err)
+			}
+			log.Info("Using the provided chequebook", "chequebookAddr", address)
+		} else {
+			err := s.swap.Deploy(context.Background(), s.backend, s.config.Path)
+			if err != nil {
+				return fmt.Errorf("Unable to deploy swap contract: %v", err)
+			}
+			log.Info("SWAP contract deployed", "contract info", s.swap.DeploySuccess())
```

Comment:
Please add to the text "New" (so it reads "New SWAP contract deployed")

Reply:
done